renderdoc/driver/vulkan/vk_layer.cpp

```diff
@@ -413,6 +413,16 @@ VK_LAYER_RENDERDOC_CaptureEnumerateInstanceExtensionProperties(
 VK_LAYER_EXPORT VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL
 VK_LAYER_RENDERDOC_CaptureGetDeviceProcAddr(VkDevice device, const char *pName)
 {
+  if(device == VK_NULL_HANDLE || pName == NULL)
+    return NULL;
+
+  PFN_vkGetDeviceProcAddr gpa = GetDeviceDispatchTable(Unwrap(device))->GetDeviceProcAddr;
+
+  // if GPA returns NULL, that means that the standard prohibits non-NULL values for this pName
+  //   or that the return value is undefined
+  if(!gpa || !gpa(Unwrap(device), pName))
+    return NULL;
```

Comment:
This change seems wrong to me as we should not rely on upstream layers/loader to define our correctness. Aside from that the loader interface document says that layers *must not* call down the chain inside `vkGetDeviceProcAddr`.

What is the problem you're trying to solve here? The behaviour difference defined in `KHR_maintenance5` should already be correctly handled by RenderDoc.

Reply:
changed it so functions explicitly say in which version of vulkan they were integrated into core

Reply:
Version gating is already handled per extension by the condition, so I still don't see the purpose of this code change.

I asked above what the problem is that you're trying to solve, and you've not answered that question. Changing the code without answering the question or explaining yourself doesn't address this feedback.

Reply:
I'm sorry. It is just if for example `VK_KHR_dynamic_rendering` is enabled with requested instance version of 1.2. RenderDoc will resolve `vkCmdBeginRendering` but it was not core in version 1.3 so it should be NULL.